tsdb/engine/tsm1/compact.go

```diff
@@ -1073,15 +1073,16 @@ func (c *Compactor) CompactFast(tsmFiles []string, logger *zap.Logger) ([]string
 
 }
 
-// removeTmpFiles is responsible for cleaning up a compaction that
+// RemoveTmpFiles is responsible for cleaning up a compaction that
 // was started, but then abandoned before the temporary files were dealt with.
-func (c *Compactor) removeTmpFiles(files []string) error {
+func (c *Compactor) RemoveTmpFiles(files []string) error {
```

Comment:
I wonder if it would make sense to add a little warning log message here if `files <= 0` something like "RemoveTmpFiles provided with 0 files". The expectation is that this gets called with a list of files and that it should not ever be empty 🤔

Reply:
I think the repurposing of it to be called on the output from `CompactFast` and `CompactFull` means I now expect it to be called with empty file lists often. There are only a few code paths out of those (and the underlying `compact`) where `files` and `err` are both non-nil.    The new call in `engine.go` will almost always have an empty file list.